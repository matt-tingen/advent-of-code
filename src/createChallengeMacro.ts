import { getSolver } from '~/run';

export const createChallengeMacro = (year: number) =>
  createMacro(
    async (day: number, part: 'a' | 'b') => {
      const solver = await getSolver(year, day, part);
      const result = await solver();

      expect(result).toMatchSnapshot();
    },
    (provided, day, part) => provided || `Challenge ${year} ${day}${part}`,
  );
