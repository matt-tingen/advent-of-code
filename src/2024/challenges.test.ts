import { createChallengeMacro } from '~/createChallengeMacro';

const testChallenge = createChallengeMacro(2024);

run(testChallenge, 1, 'a');
run(testChallenge, 1, 'b');
