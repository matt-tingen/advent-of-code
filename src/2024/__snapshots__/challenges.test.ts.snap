// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge 2024 1a 1`] = `1341714`;

exports[`Challenge 2024 1b 1`] = `27384707`;
